Fix #bugnum: 760 Use requirements.txt for dvc requirement. Add requirements.txt to src…

welenofsky opened the pull request:
This resolves #760 

Note: I ~~did not find a existing requirements.txt inside the repo. I noticed the docs mentioning src/requirements.txt so I placed the new requirements.txt there. If this was a mistake let me know. I also~~ removed the references to manually installing DVC as suggested. I then added a note to the getting started index.md telling users this guide requires git, with a link to git-scm.org to download

jorgeorpinel commented on static/docs/tutorials/deep/preparation.md:
```diff
@@ -49,17 +49,6 @@ $ echo ".env/" >> .gitignore
-You'll need DVC on your system to continue this tutorial. Please refer to
-[Installation](/doc/install) for the easiest way to install DVC on your system.
-For this tutorial, installing DVC with [pip](https://pip.pypa.io/) is probably
-the easiest:
```
Thanks a lot @welenofsky, this would've been correct per the issue's original suggestion. However as noted in https://github.com/iterative/dvc.org/issues/760#issuecomment-548185227 we decided not to do this. I'm also updating this in https://github.com/iterative/dvc.org/pull/763/files BTW... Will think about how to use your PR though. Let me think...

jorgeorpinel replied:
See https://github.com/iterative/dvc.org/pull/761#issuecomment-548561984

jorgeorpinel commented on static/docs/tutorials/deep/preparation.md:
````diff
@@ -49,17 +49,6 @@ $ echo ".env/" >> .gitignore
-```dvc
-$ pip install dvc
-```
-
````
Actually this part I agree should be removed BTW, in fact I'm also doing so in #763.

jorgeorpinel commented on static/docs/tutorials/pipelines.md:
```diff
@@ -24,13 +24,6 @@ and reproducible way.
-> If you're using Windows, please review
-> [Running DVC on Windows](/doc/user-guide/running-dvc-on-windows) for important
-> tips to improve your experience.
-
```
There was no need to remove the note about Windows though.

jorgeorpinel commented on static/docs/tutorials/versioning.md:
```diff
@@ -26,13 +26,6 @@ model file.
-> If you're using Windows, please review
-> [Running DVC on Windows](/doc/user-guide/running-dvc-on-windows) for important
-> tips to improve your experience.
-
```
Same here.

jorgeorpinel commented:
@welenofsky basically though your changes are appropriate we ended up moving in a different direction as you can see in https://github.com/iterative/dvc.org/pull/763/commits/b7ad28ff43a6bb855ba4a6594017d66d570517a0. Sorry!

> ...then added a note to the getting started index.md telling users this guide requires git, with a link to git-scm.org to download.

This would actually be really helpful if you could help us with. I didn't see it in the changes of this PR though. Basically you would revert all your previous changes (sorry again) or even open a new PR and add notes similar to the one you can see in the link I added above, but for the Get Started index or any other docs you find that it maybe useful other than tutorials (which I've done) and Get Started (which you would do).

Lmk, I can also do this myself later if not but we love interacting with contributors! 🙂

welenofsky commented:
@jorgeorpinel Can do. Will jump on in a few hours when I get home. Do you want me to just say something along the lines of "Note: these tutorials require `git`. If you do not have git installed you may download it from https://git-scm.com" ?

jorgeorpinel commented:
Awesome! I said `You'll need [Git](https://git-scm.com) to run the commands in this tutorial.` in my changes to tutorials. Feel free to check them out: https://github.com/iterative/dvc.org/commit/b7ad28ff43a6bb855ba4a6594017d66d570517a0

shcheklein commented:
@welenofsky @jorgeorpinel any progress on this or should we just close this?

jorgeorpinel commented:
This was last talked about on Friday so let's wait a couple more days? I'm sure Justin will chime in. Thanks

welenofsky commented:
Sorry about the lag. I have updated the get-started index.md with the git instructions you requested. I have reverted the changes to all other markdown files as requested as well. Thanks for letting me help/PR this. I appreciate it.

shcheklein commented:
@jorgeorpinel is it good to merge?

jorgeorpinel commented:
Yes. Sorry for the delay on my part.

jorgeorpinel commented:
p.s. this will be further modified with #747 BTW. Just a note.

shcheklein commented:
Thanks @welenofsky !
